File: Npc/11004555.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11004555: Tristan
/// </summary>
public class _11004555 : NpcScript {
    internal _11004555(INpcScriptContext context) : base(context) {
        Id = 10;
    }

    public override bool Next(int selection = 0) {
        switch (Id) {
            case 0:
                // $script:0123111007012734$ 
                // - Anything might happen now...
                return true;
            case 10:
                // $script:0123111007012735$ 
                // - Anything might happen now...
                return true;
            default:
                return true;
        }
    }
}

File: Npc/11004555.py
""" 11004555: Tristan """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 10

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0123111007012734$
        # - Anything might happen now...
        return None # TODO

    def __10(self, pick: int) -> int:
        # $script:0123111007012735$
        # - Anything might happen now...
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (10, 0):
            return Option.CLOSE
        return Option.NONE
